apps/src/dance/lab2/ProgramExecutor.ts

```diff
@@ -114,31 +116,18 @@ export default class ProgramExecutor {
       return;
     }
 
-    const previewDraw = () => {
-      this.nativeAPI.setEffectsInPreviewMode(true);
-
-      // the user setup hook initializes effects,
-      // needs to happen in preview mode for some effects (eg, tacos)
-      if (!this.hooks.runUserSetup) {
-        return;
-      }
-      this.hooks.runUserSetup();
-
-      // redraw() (rather than draw()) is p5's recommended way
-      // of drawing once.
-      this.nativeAPI.p5_.redraw();
-
-      this.nativeAPI.setEffectsInPreviewMode(false);
-    };
-
-    // This is the mechanism p5 uses to queue draws,
-    // so we do the same so we end up after any queued draws.
-    window.requestAnimationFrame(previewDraw);
+    this.hooks.runUserSetup();
+    // TODO: We are calling livePreview() here since this is currently only used by
+    // the AI Dance modal. When this is integrated with Lab2, we should probably also
+    // support static previews.
+    this.nativeAPI.livePreview(songMetadata);
+    this.livePreviewActive = true;
   }
 
   reset() {
     Sounds.getSingleton().stopAllAudio();
     this.nativeAPI.reset();
+    this.livePreviewActive = false;
```

Comment:
Does `livePreviewActive` get set to false at some point (I was imagining when the modal is closed?) reset() looks like it only gets called when starting a preview? Not sure if it matters

Reply:
Yeah, it does get called in reset() but you're right that we don't call reset explicitly. We do call `destroy()` when unmounting the preview, so I'll add it there as well to be safe.

For context - the flag mainly exists to avoid calling the `runUserEvents` function since there shouldn't actually be any user events occurring during the preview.

Reply:
Ok cool -- maybe I'm wondering more broadly about stopping the preview generally. Since we don't call `reset()`, we never actually stop the p5 loop? Does that matter?

Reply:
Well we do call `destroy()` on unmount which calls `teardown()` in the native API layer, which in turn calls `p5_.remove()`, which internally sets `this._loop` to `false` (same as `p5_.noLoop()`), so I think we should be good?

Reply:
Oh cool, I did not realize returning a function in useEffect works as cleanup!